### PR TITLE
Add lsofi types

### PR DESCRIPTION
**Pull request by oof2win2**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [X] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by peterblazejewicz — CHANGES_REQUESTED** [review]
Please use default formatting from the project (run prettier as per readme).
Please also add a test file back, it could be as simply as code from usage:
https://github.com/marionebl/lsofi#usage

**Comment by oof2win2** [comment]
Am I supposed to write functional tests or just tests to check the type?
This would check the type to be what the function returns
```ts
import lsofi from "lsofi";
lsofi(12345) // $ExpectType Promise<number|null>
```

Whilst this would rather check that the function works in the sense that it returns a number if the port is used by a process.
```ts
import lsofi from "lsofi";
import http from "http";

const tcpServer = http.createServer().listen();
tcpServer.once("listening", async () => {
	const address = tcpServer.address();
	if (typeof address === "string") {
		// idk what to do here since nodejs docs don't show the format of it
	} else {
		// $ExpectType number
		await lsofi(address.port); // $ExpectType number
	}
})
```

**Comment by peterblazejewicz** [comment]
> This would check the type to be what the function returns

this works best IMO

**Comment by oof2win2** [comment]
Added the tests, can't run the workflow though.

**Comment by oof2win2** [comment]
My issue is that this error is thrown and I don't think this is right.
```
ERROR: 4:1  expect  TypeScript@4.4 expected type to be:
  Promise<number|null>
got:
  Promise<number | null>
```

File: types/lsofi/lsofi-tests.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import lsofi from "lsofi";
```
Comment (peterblazejewicz) [inline types/lsofi/lsofi-tests.ts:1]:
```suggestion
import lsofi = require("lsofi");
```

File: types/lsofi/tsconfig.json [hunk 2]
```diff
@@ -1,16 +1,25 @@
 {
 	"compilerOptions": {
+		"esModuleInterop": true,
```
Comment (peterblazejewicz) [inline types/lsofi/tsconfig.json:3]:
```suggestion
```

File: types/lsofi/tsconfig.json [hunk 3]
```diff
@@ -1,16 +1,25 @@
...
 		"baseUrl": "../",
-		"typeRoots": ["../"],
+		"typeRoots": [
+			"../"
+		],
 		"types": [],
-		"lib": ["ES6"],
+		"lib": [
+			"ES6",
+			"DOM"
```
Comment (peterblazejewicz) [inline types/lsofi/tsconfig.json:14]:
```suggestion
			"ES6"
```
there is nothing from `lib.dom` used for this to be required, am I correct?

**Comment by peterblazejewicz** [comment]
> Please use default formatting from the project (run prettier as per readme).

```bash
npx prettier --write .\types\lsofi\*
```

File: types/lsofi/tsconfig.json (hunk 3 above)
Comment (oof2win2) [reply types/lsofi/tsconfig.json:14]: Yep, that was old from `http`.

**Comment by oof2win2** [comment]
Closed because IDK how to fix this.

**Pull request closed without merging**